sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2293,9 +2293,10 @@ class config_2d_mem_access {
   /// </summary>
   /// <returns>surface base address</returns>
   T *get_data_pointer() const {
-    return (T *)(const_cast<config_2d_mem_access *>(this)
-                     ->payload_data.template bit_cast_view<uint64_t>()
-                     .template select<1, 1>(0));
+    return (
+        T *)((uint64_t)(const_cast<config_2d_mem_access *>(this)
+                            ->payload_data.template bit_cast_view<uint64_t>()
+                            .template select<1, 1>(0)[0]));
```

Comment:
Do you really need 'select' that selects 1 element if you then use operator[] right after it?
I believe this code can be shorter and still work:
```suggestion
    return (
        T *)((uint64_t)(const_cast<config_2d_mem_access *>(this)
                            ->payload_data.template bit_cast_view<uint64_t>()[0]));
```